torch/fx/experimental/sym_node.py

```diff
@@ -448,6 +448,125 @@ def is_constant(self):
         return False
 
 
+def _tid_eq(a, b):
+    from torch.nested._internal import union_find
+    uf = union_find.get_union_find()
+    a_canonical = uf._union_find_int.find(a)
+    b_canonical = uf._union_find_int.find(b)
+    return a_canonical == b_canonical
+
+
+def _eq(lhs, rhs) -> bool:
+    return (
+        isinstance(lhs, NestedIntNode)
+        and isinstance(rhs, NestedIntNode)
+        and _tid_eq(lhs.t_id, rhs.t_id)
+        and lhs.coeff == rhs.coeff
+    )
+
+
+def _ge(lhs, rhs) -> bool:
+    if isinstance(rhs, NestedIntNode) and isinstance(lhs, NestedIntNode):
+        if _tid_eq(lhs.t_id, rhs.t_id):
+            return lhs.coeff >= rhs.coeff
+        raise ValueError("ge: relation is indeterminate")
+    elif isinstance(lhs, NestedIntNode):
+        if rhs.is_constant() and rhs.constant_int() <= 2:
+            return True
+        raise ValueError("ge: relation is indeterminate")
+    elif isinstance(rhs, NestedIntNode):
+        if lhs.is_constant() and lhs.constant_int() < 2:
+            return False
+        raise ValueError("ge: relation is indeterminate")
+    else:
+        raise ValueError(f"inputs unsupported")
+
+
+class NestedIntNode:
```

Comment:
request for documentation describing the purpose of this / some description of its workings

Reply:
How do you feel about just adding a code pointer to the docs for the cpp version for now aten/src/ATen/core/NestedIntSymNodeImpl.h, and then moving that over if that is deleted?